File: test_data/code/java/1844-replace-all-digits-with-characters.java
class Solution {
    public String replaceDigits(String s) {
        StringBuilder result = new StringBuilder();
        for (int i = 0; i < s.length(); i++) {
            char c;
            if (i % 2 == 1) {
                c = (char) ((s.charAt(i - 1) - 'a' + (s.charAt(i) - '0')) % 26 + 'a');
            } else {
                c = s.charAt(i);
            }
            result.append(c);
        }
        return result.toString();
    }
}

File: test_data/code/python/1844-replace-all-digits-with-characters.py
class Solution:
    def replaceDigits(self, s: str) -> str:
        result = list(s)
        for i in range(1, len(s), 2):
            prev, curr = result[i - 1], result[i]
            result[i] = chr(ord(prev) + int(curr))
        return ''.join(result)